src/ArraySerializable.php

```diff
@@ -16,13 +16,11 @@ class ArraySerializable extends AbstractHydrator
      *
      * Extracts values via the object's getArrayCopy() method.
      *
-     * @param  object $object
-     * @return array
      * @throws Exception\BadMethodCallException for an $object not implementing getArrayCopy()
      */
-    public function extract($object)
+    public function extract(object $object) : array
     {
-        if (! is_callable([$object, 'getArrayCopy'])) {
+        if (! method_exists($object, 'getArrayCopy')) {
```

Comment:
Huge warning! `method_exists()` also checks private API: `is_callable()` was correct here. A test is needed here, since the regression wasn't noticed.

Reply:
I made these changes based on phpstan, which was reporting the call would always return `false`. I've updated these to be a combination of `method_exists` + `is_callable`, and that removes the phpstan warnings.